vowpalwabbit/core/src/reductions/cb/cb_explore_adf_large_action_space.cc

```diff
@@ -105,24 +105,23 @@ void cb_explore_adf_large_action_space<impl_detail>::update_example_prediction(V
       return;
     }
 
-    _spanner_state.compute_spanner(U, _d);
+    _spanner_state.compute_spanner(U, _d, shrink_factors);
+    _spanner_state.compute_spanner1(U, _d);
+
     assert(_spanner_state._spanner_bitvec.size() == preds.size());
   }
   else
   {
     // When the number of actions is not larger than d, all actions are selected.
-    _spanner_state._spanner_bitvec.clear();
-    _spanner_state._spanner_bitvec.resize(preds.size(), true);
+    return;
   }
 
   // Keep only the actions in the spanner so they can be fed into the e-greedy or squarecb reductions.
   // Removed actions will be added back with zero probabilities in the cb_actions_mask reduction later
   // if the --full_predictions flag is supplied.
-  size_t index = 0;
   for (auto it = preds.begin(); it != preds.end(); it++)
   {
-    if (!_spanner_state._spanner_bitvec[index]) { preds.erase(it--); }
-    index++;
+    if (!_spanner_state._spanner_bitvec[it->action]) { preds.erase(it--); }
```

Comment:
`it--` from begin() is undefined, although it later increases. Better make it safer.

Reply:
replaced